modules/tinymce/src/core/main/ts/lookup/UserLookup.ts

```diff
@@ -0,0 +1,196 @@
+ *
+ * @class tinymce.UserLookup
+ * @example
+ * // Retrieve the current user ID
+ * tinymce.activeEditor.userLookup.getCurrentUserId();
+ *
+ * // Fetch user information by IDs which returns array of promises
+ * const promises = tinymce.activeEditor.userLookup.fetchUsers(['user-1', 'user-2']);
+ * Promise.all(promises).then((users) => {
+ *   users.forEach(user => console.log('User found:', user));
+ * }).catch((error) => {
+ *   console.error('Error fetching users:', error);
+ * });
+ */
+
+type UserId = string;
+
+export interface User {
+  id: UserId;
+  name?: string;
+  avatar?: string;
+  description?: string;
+  custom?: Record<string, any>;
+}
+
+export interface UserLookup {
+  /**
+   * Retrieves the current user ID from the editor.
+   *
+   * @method getCurrentUserId
+   * @return {string} The current user ID.
+   */
+  getCurrentUserId: () => UserId;
+
+  /**
+   * Fetches user information using a provided array of userIds.
+   *
+   * @method fetchUsers
+   * @param {string[]} userIds - A list of user IDs to fetch information for.
+   * @throws {Error} Throws an error if any of the user IDs are invalid or if the fetch fails.
+   * @return {Promise<User>[]} A promise that resolves to an array of users and information about them.
+   */
+  fetchUsers: (userIds: UserId[]) => Promise<User>[];
+}
+
+const userSchema = StructureSchema.objOf([
+  FieldSchema.required('id'),
+  FieldSchema.optionString('name'),
+  FieldSchema.optionString('avatar'),
+  FieldSchema.optionString('description'),
+  FieldSchema.option('custom')
+]);
+
+const transformResult = (user: any): User => {
+  const customValue = user.custom?.getOrUndefined();
+  return {
+    id: user.id,
+    name: user.name?.getOrUndefined(),
+    avatar: user.avatar?.getOrUndefined(),
+    description: user.description?.getOrUndefined(),
+    ...(customValue !== undefined && { custom: customValue })
+  };
+};
+
+const validateResponse = (items: unknown): User[] => {
+  if (!Array.isArray(items)) {
+    throw new Error('fetch_users must return an array');
+  }
+
+  const results = Arr.map(items, (item) =>
+    StructureSchema.asRaw<User>('Invalid user object', userSchema, item)
+  );
+
+  const { errors, values } = Results.partition(results);
+
+  if (errors.length > 0) {
+    const formattedErrors = Arr.map(errors, (error, idx) =>
+      `User at index ${idx}: ${StructureSchema.formatError(error)}`
+    );
+
+    // eslint-disable-next-line no-console
+    console.warn('User validation errors:\n' + formattedErrors.join('\n'));
+  }
+
+  return Arr.map(values, (result) => transformResult(result));
+};
+
+const UserLookup = (editor: Editor): UserLookup => {
+  const userCache = new Map<UserId, Promise<User>>();
+  const pendingResolvers = new Map<UserId, {
+    resolve: (user: User) => void;
+    reject: (error: unknown) => void;
+  }>();
+
+  const lookup = (userId: UserId) =>
+    Optional.from(userCache.get(userId)).map(Fun.identity);
```

Comment:
uhhh `.map(Fun.identity)` does literally nothing 🤔 what were you intending to happen here?
```suggestion
    Optional.from(userCache.get(userId));
```